astropy/modeling/fitting.py

```diff
@@ -742,9 +743,14 @@ def __call__(self, model, x, y, z=None, weights=None,
         from scipy import optimize
 
         model_copy = _validate_model(model, self.supported_constraints)
-        farg = (model_copy, weights, ) + _convert_input(x, y, z)
-
-        if model_copy.fit_deriv is None or estimate_jacobian:
+        n_models = len(model_copy)
+        model_set_axis = False if n_models > 1 else model_copy.model_set_axis
```

Comment:
So when using model sets I need to pass `model_set_axis = False` for the model evaluation. Is this the good way to proceed ?

Reply:
Yes, I think once #7222 is merged, you always want to use `model_set_axis=False` when evaluating models (in `objective_function`) on the single set of `x`/`y` co-ordinates that gets passed to fitters. Also, I think `model_set_axis` is always `False` when `n_models == 1`, so this condition seems redundant and can just be `False`? I don't think this needs to happen in `__call__` itself.